### PR TITLE
DR-2763 Make sure that snapshots don't union too much

### PR DESCRIPTION
**Pull request by nmalfroy**
This limits the number of tables being unioned together but loading the rowids during the snapshotting process in chunks of 10 tables

### CODE REVIEW

**Review by okotsopoulos — APPROVED** [review]
I think this looks good, but is there an opportunity to add test coverage to verify behavior?

**Comment by nmalfroy** [comment]
I was able to confirm with a manual test: creating a snapshot with 100 test gives me the same error that users were seeing without this fix and then this fix it works!

File: src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQuerySnapshotPdao.java [hunk 1]
```diff
@@ -301,23 +302,26 @@ public void createSnapshotWithLiveViews(Snapshot snapshot, Dataset dataset, Inst
     List<DatasetTable> tables = dataset.getTables();
 
     // create a snapshot table based on the live view data row ids
-    String liveViewTables =
-        createSnapshotTableFromLiveViews(
-            datasetBigQueryProject, tables, datasetBqDatasetName, filterBefore);
+    // Batch into smaller groups of tables to avoid BQ complexity errors
+    for (var tablesBatch : ListUtils.partition(tables, TABLE_UNION_BATCH_SIZE)) {
+      String liveViewTables =
+          createSnapshotTableFromLiveViews(
```
Comment (otchet-broad) [inline src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQuerySnapshotPdao.java:308]: Do we need to consider that a failure case can now occur where only some of the inserts are successful?  Should the set of insert operations now be included in a transaction so rollback can occur if one of the inserts fails?

Comment (otchet-broad) [reply src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQuerySnapshotPdao.java:308]: @nmalfroy confirmed the flight will unwind this if any individual statements fail.  No need to wrap in a transaction.

**Review by otchet-broad — APPROVED** [review]
@nmalfroy - based on our discussion regarding transactions, approved.

**Pull request merged**